server.js

```diff
@@ -0,0 +1,386 @@
+      </p>
+      <p><a href="/">back</a></p>
+    </body>
+  </html>
+  `;
+        fs.writeFile(
+          "./public/" + formDataObject["elementName"].toLowerCase() + ".html",
+          newElement,
+          function(error) {
+            if (error) throw error;
+            console.log("File created successfully");
+          }
+        );
+        // Send success response
+        request.on("end", function() {
+          response.writeHead(200, { "Content-Type": "application/json" });
+          let responseBody = { success: "true" };
+          response.end(JSON.stringify(responseBody));
+        });
+      }
+    });
+  }
+
+  // DELETE REQUESTS
+
+  if (request.method === "DELETE") {
+    console.log(request.method + " " + request.url);
+    // Return 500 if file does not exist
+    if (!fs.existsSync("./public" + request.url)) {
+      console.log("resource " + request.url + " does not exist");
+      response.writeHead(500, { "Content-Type": "application/json" });
+      let responseBody = {
+        error: "resource " + request.url + " does not exist"
+      };
+      response.end(JSON.stringify(responseBody));
+      // Disallow deletion of essential files
+    } else if (
+      request.url === "/index.html" ||
+      request.url === "/404.html" ||
+      request.url === "/css/styles.css" ||
+      request.url === "/hydrogen.html" ||
+      request.url === "/helium.html"
+    ) {
+      console.log("resource " + request.url + " cannot be deleted");
+      response.writeHead(500, { "Content-Type": "application/json" });
+      let responseBody = {
+        error: "resource " + request.url + " cannot be deleted"
+      };
+      response.end(JSON.stringify(responseBody));
+    } else {
+      // Delete element html file
+      fs.unlinkSync("./public" + request.url);
+      console.log(request.url + " deleted successfully");
+      // Update index.html
+      fs.readFile("./public/index.html", function(error, data) {
+        if (error) throw error;
+        // Find the element to delete
+        let indexHTML = data.toString();
+        let split = indexHTML.split("\n");
+        for (let i = 0; i < split.length; i++) {
+          if (split[i].includes(request.url) === true) {
+            // Splice that element from the list
+            split.splice(i - 1, 3);
+          }
+        }
+        // Count the remaining elements
+        let count = 0;
+        for (let i = 0; i < split.length; i++) {
+          if (split[i].includes("<li>") === true) {
+            count++;
+          }
+        }
+        // Update <h3> element count
+        for (let i = 0; i < split.length; i++) {
+          if (split[i].includes("<h3>") === true) {
+            split[i] = "    <h3>There are " + count + "</h3>";
+          }
+        }
+        // Write to index.html
+        indexHTML = split.join("\n");
+        fs.writeFile("./public/index.html", indexHTML, function(error) {
+          if (error) throw error;
```

Comment:
Errors should be handled properly, they should send back a response with a appropriate status code and message.